gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -416,7 +418,14 @@ private static Function convertToQueryFunction(
         }
       default:
         {
-          builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
+          String functionName = function.getFunction().name();
+          if (DISTINCT_ARRAY.name().equals(functionName)) {
```

Comment:
Can we just add it as a switch case above and throw the exception?

Reply:
Right. Updated it now.